src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/repoting/VisitsReportingService.kt

```diff
@@ -0,0 +1,160 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.service.repoting
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.PrisonDto
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.reporting.SessionVisitCountsDto
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.sessions.SessionScheduleDto
+import uk.gov.justice.digital.hmpps.visitscheduler.model.VisitRestriction
+import uk.gov.justice.digital.hmpps.visitscheduler.model.VisitStatus
+import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.projections.VisitRestrictionStats
+import uk.gov.justice.digital.hmpps.visitscheduler.repository.VisitRepository
+import uk.gov.justice.digital.hmpps.visitscheduler.service.PrisonConfigService
+import uk.gov.justice.digital.hmpps.visitscheduler.service.SessionService
+import java.time.LocalDate
+
+@Service
+class VisitsReportingService(
+  private val prisonConfigService: PrisonConfigService,
+  private val sessionService: SessionService,
+  private val visitRepository: VisitRepository,
+) {
+  fun getVisitCountsBySession(reportDate: LocalDate): List<SessionVisitCountsDto> {
+    return getSessionsReport(reportDate)
+  }
+
+  private fun getSessionsReport(reportDate: LocalDate): List<SessionVisitCountsDto> {
+    val prisons = getAllActivePrisons()
```

Comment:
Am I correct in thinking you are only interested in prison that are not exclude by the report date, if so I would create a new method in the prisonConfigService that returns prison that are valid for the report date.

Reply:
I need all prisons but if the date is excluded or there are no sessions we are setting flag so we can filter by prison and use them on reports